setuptools/command/build_py.py

```diff
@@ -134,10 +135,20 @@ def analyze_manifest(self):
             d, f = os.path.split(assert_relative(path))
             prev = None
             oldf = f
+
+            # Climb the hierarchy to find a parent package included in the distribution
             while d and d != prev and d not in src_dirs:
+                if has_valid_modules(d):
+                    # The given directory contains Python modules, and therefore can be
+                    # considered a package (or namespace) itself, not a simple container
+                    # for data files.
+                    # If the intention is to include it in the distribution, then it
+                    # should be added to `packages`, and therefore appear in `src_dirs`.
+                    break
                 prev = d
                 d, df = os.path.split(d)
                 f = os.path.join(df, f)
+
             if d in src_dirs:
                 if path.endswith('.py') and f == oldf:
```

Comment:
This `f == oldf` condition is a bit odd.
According to the git history ( https://github.com/pypa/setuptools/commit/f01c17e9e6301a72c54a0b3a5b9d1e21449e4e51), the original intention of this code was to:

> Allow .py files found by the include_package_data option to be automatically included.

To me that sounds a little bit as a misuse. My opinion is that modules should not be affected by `include_package_data`...

Reply:
All of the issues and PRs were migrated from Bitbucket, but that commit comes from when Setuptools was hosted on Sourceforge and Subversion, where issues and merges weren't tracked as easily, so the commit history is the best we have.